app/models/similarity/unit_similarity_module.rb

```diff
@@ -222,4 +235,51 @@ def add_done_files_for_plagiarism_check_of(task_definition, tmp_path, to_check,
 
     self
   end
+
+  # JPLAG Function - extracts "done" files for each task and packages them into a directory for JPLAG to run on
+  def run_jplag_on_done_files(task_definition, tasks_dir, tasks_with_files, unit_code)
+    type_data = task_definition.moss_language.split
+    return if type_data.nil? || (type_data.length != 2) || (type_data[0] != 'moss')
+
+    # get each code file for each task
+    task_definition.upload_requirements.each_with_index do |upreq, idx|
+      # only check code files marked for similarity checks
+      next unless upreq['type'] == 'code' && upreq['tii_check']
+
+      pattern = task_definition.glob_for_upload_requirement(idx)
+
+      tasks_with_files.each do |t|
+        t.extract_file_from_done(tasks_dir, pattern, ->(_task, to_path, name) { File.join(to_path.to_s, t.student.username.to_s, name.to_s) })
+      end
+      puts "Starting JPLAG container to run on #{tasks_dir}"
+      tasks_dir_split = tasks_dir.to_s.split('/workspace/doubtfire-api')[1]
```

Comment:
hardcoded dev container path... won't work in production

Reply:
Have changed to Rails.root